### ISSUE TITLE
Add arrays

### ISSUE THREAD

**Issue by treeowl** [fumieval/witherable#56]
Add `Array` from `array`, and try to push instances into `primitive` for `Array` and `SmallArray` there.

**Comment by fumieval** [comment]
Yes, ultimately witherable-class should only depend on packages that ship with GHC.
